### DISCUSSION TITLE
Is there python code for some classical planning algorithms like UCPOP and GraphPlan available on the Internet

### DISCUSSION THREAD

**Discussion by Sino-Huang** [AI-Planning/pddl#124] [Q&A]
Hello, I am wondering if we can find python code for some classical planning algorithms like UCPOP and GraphPlan available on the Internet. If not I am going to implement them by myself and leverage this pddl package to encode the pddl snippets

**Comment by WeihangGuo** [comment]
I am also looking for those solvers. Have you found any of it? thank you!!

**Reply by haz** [reply]
It would certainly be nice to see some of these concepts implemented over the pddl lib! But I'm not aware of anything concrete...there may be something in the AIMA book implementations on planning, and this project is the closest we have to an approachable python implementation of a planner: https://github.com/aibasel/pyperplan

**Reply by Tass0sm** [reply]
A basic implementation of GraphPlan and POP is given here: https://github.com/aimacode/aima-python/blob/master/planning.py

**Reply by Tass0sm** [reply]
I'm also working on an implementation [here](https://github.com/Tass0sm/ucpop) based on the description of POP and UCPOP in [this paper](https://homes.cs.washington.edu/~weld/papers/pi.pdf)
